pkg/controller/master/virtualmachine/vm_controller.go

```diff
@@ -0,0 +1,100 @@
+package virtualmachine
+
+import (
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	kubevirtapis "kubevirt.io/client-go/api/v1alpha3"
+
+	cdictrl "github.com/rancher/harvester/pkg/generated/controllers/cdi.kubevirt.io/v1beta1"
+)
+
+type VMController struct {
+	// dv ctrl suite
+	dataVolumeClient cdictrl.DataVolumeClient
+	dataVolumeCache  cdictrl.DataVolumeCache
+}
+
+// RefOutTreeDataVolumes records the target VirtualMachine as the owner of the out-tree DataVolumes.
+func (h *VMController) RefOutTreeDataVolumes(_ string, vm *kubevirtapis.VirtualMachine) (*kubevirtapis.VirtualMachine, error) {
```

Comment:
prefer to call it `SetOwnerOfOutTreeDataVolumes`

Reply:
Updated